cli/src/main/java/org/jboss/as/cli/handlers/DeploymentHandler.java

```diff
@@ -117,8 +131,60 @@ protected ModelNode buildReadDeploymentResourceRequest(String name) {
         return request;
     }
 
-    protected MountHandle extractArchive(File archive, TempFileProvider tempFileProvider) throws IOException {
-        return ((MountHandle)VFS.mountZipExpanded(archive, VFS.getChild("cli"), tempFileProvider));
+    static File extractArchive(final File cliArchive) throws IOException {
+        final ZipInputStream zis = new ZipInputStream(new FileInputStream(cliArchive));
+        ZipEntry entry;
+        File newFile;
+        final File targetDir = new File(CLI_TMP_DIR, cliArchive.getName() + System.currentTimeMillis());
+        try {
+            while ((entry = zis.getNextEntry()) != null) {
+                String fileName = entry.getName();
+                // create directory structure
+                newFile = new File(targetDir + File.separator + fileName);
+                new File(newFile.getParent()).mkdirs();
+                // extract zip
+                if (!entry.isDirectory()) {
+                    // extract zip entry to the disk
+                    FileOutputStream fos = null;
+                    try {
+                        fos = new FileOutputStream(newFile);
+                        copy(zis, fos);
+                    } finally {
+                        safeClose(fos);
+                    }
+                }
+            }
+        } finally {
+            safeClose(zis);
```

Comment:
this can be eliminated if you use try with resources
something like
try (final ZipInputStream zis = new ZipInputStream(new FileInputStream(cliArchive))){
...
}